rudder-webapp/SOURCES/rudder-node-to-relay

```diff
@@ -57,201 +76,25 @@ fi
-  LDAP_USER=$(grep "^rootdn" /opt/rudder/etc/openldap/slapd.conf | sed "s/\w*\s*['\"]\?\([^\"']*\)['\"]\?$/\1/")
-  LDAP_PASSWORD=$(grep "^rootpw" /opt/rudder/etc/openldap/slapd.conf | sed "s/\w*\s*['\"]\?\([^\"']*\)['\"]\?$/\1/")
-  LDAP_SERVER='localhost'
-  LDAP_PORT='389'
-fi
-
-# Commands
-LDAP_PARAMETERS="-H ldap://${LDAP_SERVER}:${LDAP_PORT}/ -D ${LDAP_USER} -w ${LDAP_PASSWORD} -x"
-
-JETTY_INIT="service rudder-jetty"
-
-LDAPSEARCH="ldapsearch ${LDAP_PARAMETERS} -LLL"
-LDAPMODRDN="ldapmodrdn ${LDAP_PARAMETERS}"
-LDAPADD="ldapadd ${LDAP_PARAMETERS}"
-LDAPDELETE="ldapdelete ${LDAP_PARAMETERS}"
-
-PSQL="psql -q -h ${SQL_SERVER} -p ${SQL_PORT} -U ${SQL_USER}"
-
-#######################################################################################################################
-## Delete the old entries
-#######################################################################################################################
-
-# We need to read the actual UUID from LDAP to make sure we get the same case. LDAP is insensitive to this, but the
-# Rudder web interface will misbehave if the wrong case is used
-RELAY_UUID=$(${LDAPSEARCH} -s base -b "nodeId=${RELAY_UUID},ou=Nodes,ou=Accepted Inventories,ou=Inventories,cn=rudder-configuration" nodeId 2>/dev/null | grep "^nodeId:" | sed 's%nodeId: %%' || true)
-EXISTING_NODEID=$(${LDAPSEARCH} -s base -b "nodeGroupId=hasPolicyServer-${RELAY_UUID},groupCategoryId=SystemGroups,groupCategoryId=GroupRoot,ou=Rudder,cn=rudder-configuration" nodeId 2>/dev/null | grep "^nodeId:" || true)
-EXISTING_NETWORKS=$(${LDAPSEARCH} -s base -b "directiveId=common-root,activeTechniqueId=common,techniqueCategoryId=Rudder Internal,techniqueCategoryId=Active Techniques,ou=Rudder,cn=rudder-configuration" directiveVariable 2>/dev/null | grep "^directiveVariable: ALLOWEDNETWORK\[" || true)
-EXISTING_TYPE=$(${LDAPSEARCH} -s base -b "nodeId=${RELAY_UUID},ou=Nodes,cn=rudder-configuration" isSystem 2>/dev/null | grep "^isSystem:" || true)
-RELAY_HOSTNAME=$(${LDAPSEARCH} -s base -b "nodeId=${RELAY_UUID},ou=Nodes,ou=Accepted Inventories,ou=Inventories,cn=rudder-configuration" nodeHostname 2>/dev/null | grep "^nodeHostname:" | sed 's%nodeHostname: %%' || true)
-
-if [ "${RELAY_HOSTNAME}" = "" ]; then
-	${ECHO} "ERROR: Cannot find the machine (inventory not found), aborting."
-	exit 1
-fi
-
-if [ "${EXISTING_TYPE}" = "isSystem: FALSE" ]; then
-	${ECHO} -e "INFO: The machine is identified as a regular node, whose name is ${RELAY_HOSTNAME}\n"
-elif [ "${EXISTING_TYPE}" = "isSystem: TRUE" ]; then
-	${ECHO} "INFO: The machine ${RELAY_HOSTNAME} is already either a Rudder relay or root server."
+# Scale-out-relay API call (must have the plugin installed)
+API_URL="https://127.0.0.1/rudder/api/latest/scaleoutrelay/promote/${RELAY_UUID}"
+TOKEN_SYSTEM=$(cat /var/rudder/run/api-token)
+curl_command="${DOWNLOAD_COMMAND} --header \"X-API-Token: ${TOKEN_SYSTEM}\" --request POST \"${API_URL}\""
+response=$(eval ${curl_command} )
```

Comment:
try avoiding eval as much as possible, because it is a frequent source of bugs
Here you could use : response=$(${DOWNLOAD_COMMAND} --header "X-API-Token: ${TOKEN_SYSTEM}" --request POST "${API_URL}")
provided that you make a little change to DOWNLOAD_COMMAND

Reply:
you also need to check $? after this line in case of an error in the http query itself